e2e/stack_monitoring_test.go

```diff
@@ -0,0 +1,614 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/Jeffail/gabs/v2"
+	"github.com/cucumber/godog"
+	messages "github.com/cucumber/messages-go/v10"
+	"github.com/elastic/e2e-testing/cli/config"
+	"github.com/elastic/e2e-testing/cli/services"
+	"github.com/nsf/jsondiff"
+	log "github.com/sirupsen/logrus"
+)
+
+// StackMonitoringTestSuite represents a test suite for the stack monitoring parity tests
+type StackMonitoringTestSuite struct {
+	Env       map[string]string
+	IndexName string
+	Port      string
+	Product   string
+	// collection method hits
+	collectionHits map[string]map[string]interface{}
+	// extra fields used during assertions
+	allowedDeletionsExtra  []string
+	allowedInsertionsExtra []string
+	handleSpecialCases     func(docType string, legacy *gabs.Container, metricbeat *gabs.Container) error
+}
+
+// @collectionMethod1 the collection method to be used. Valid values: legacy, metricbeat
+// @collectionMethod2 the collection method to be used. Valid values: legacy, metricbeat
+func (sm *StackMonitoringTestSuite) checkDocumentsStructure(
+	collectionMethod1 string, collectionMethod2 string) error {
+
+	log.Debugf("Compare the structure of the %s documents with the structure of the %s documents", collectionMethod1, collectionMethod2)
+
+	return assertHitsEqualStructure(sm, collectionMethod1, collectionMethod2)
+}
+
+// checkProduct sets product name, in lowercase, setting the product port based on a validation:
+// if the product is not supported, then an error is thrown
+func (sm *StackMonitoringTestSuite) checkProduct(product string, collectionMethod string) error {
+	sm.Product = strings.ToLower(product)
+
+	env := map[string]string{
+		"monitoringEsHost": "monitoringEs", // monitoring elasticsearch service name
+		"monitoringEsPort": "9200",         // monitoring elasticsearch port
+	}
+
+	log.Debugf("Enabling %s collection, sending %s metrics to the monitoring instance", collectionMethod, sm.Product)
+
+	productIndexID := sm.Product
+
+	switch {
+	case sm.Product == "elasticsearch":
+		sm.Port = strconv.Itoa(9201)
```

Comment:
What is the rationale behind setting the port here (and same question for other products below as well)?

Reply:
This was one of the initial things I did when started this PR, and thinking about it now, it makes no sense. I'll remove them until needed

Reply:
BTW, I realised that for the Elasticsearch test, the port it's needed, as there is an env variable (product + "Port": sm.Port) that binds the port to the docker host, and that port (9200) is already in use by the elasticsearch stack instance